### PR TITLE
Treeselectmenu vanillajs

### PR DESCRIPTION
**Pull request by Supun94**
Pull Request for Issue #13 .

VanillaJS conversion for Treeselectmenu.

### CODE REVIEW

File: media/system/js/treeselectmenu.js [hunk 1]
```diff
@@ -2,117 +2,136 @@
...
 
-		if ($li.find('ul.treeselect-sub').length) {
+		if (li.querySelector('ul.treeselect-sub').length) {
 			// Add classes to Expand/Collapse icons
-			$li.find('span.icon-').addClass('treeselect-toggle fa-chevron-down');
+			li.querySelectorAll('span.icon-').classList.add('treeselect-toggle fa-chevron-down');
 
 			// Append drop down menu in nodes
-			$div.find('label:first').after(treeselectmenu);
+			var element = div.querySelector('label:first');
```
Comment (Ruchiranga) [inline media/system/js/treeselectmenu.js]: shouldn't `querySelectorAll` be used here?

Comment (Ruchiranga) [inline media/system/js/treeselectmenu.js]: and once again not sure about the validity of the selector string

File: media/system/js/treeselectmenu.js [hunk 2]
```diff
@@ -2,117 +2,136 @@
...
+			element.parentNode.insertBefore(treeselectmenu, element);
```
Comment (Ruchiranga) [inline media/system/js/treeselectmenu.js]: Refer this
https://stackoverflow.com/questions/4793604/how-to-do-insert-after-in-javascript-without-using-a-library

I think it should be `element.parentNode.insertBefore(treeselectmenu, element.nextSibling);`

File: media/system/js/treeselectmenu.js [hunk 3]
```diff
@@ -2,117 +2,136 @@
...
 
-			if (!$li.find('ul.treeselect-sub ul.treeselect-sub').length) {
-				$li.find('div.treeselect-menu-expand').remove();
+			if (!li.querySelector('ul.treeselect-sub ul.treeselect-sub').length) {
+				li.querySelector('div.treeselect-menu-expand').classList.remove();
```
Comment (Ruchiranga) [inline media/system/js/treeselectmenu.js]: refer https://api.jquery.com/remove/
`.remove()` removes the element. Not the classses.

**Comment by Supun94** [comment]
Unexpected build failure in drone and sent a #44 PR with same changes. Please review

**Pull request closed without merging**